docs/design/da-pki-revocation-design.md

```diff
@@ -0,0 +1,213 @@
+
+3. If `IssuerSubjectKeyID` is not equal to the `Subject Key Identifier` of `CRLSignerCertificate`, the transaction is considered invalid and rejected.
+
+4. If `RevocationType` is 1 (RFC5280 CRL), then `DataFileSize`, `DataDigest`, `DataDigestType` must be empty.
+
+   Otherwise, it may cause confusion. Certificates to be revoked are added to the content of CRL URL without a need to send a transaction to the ledger.
+   So, someone may look at the DCL entry, see that a file size is specified, then go to the URL, and see that the actual size there is different.
+   It may cause a question if the data is valid, and why a different file size is in DCL.
+
+5. `ProductID` is expected to be `unit16`, not `unit32` as in other places.
+
+
+## A need for Proof-of-possession of `CRLSignerCertificate` key
+No additional proof-of-possession is required because 
+- The transaction is already signed by the Vendor account key, so only a trusted Vendor can send it.
+- DCL checks that `CRLSignerCertificate` is either a valid PAI (signed by a trusted PAA from DCL) or a PAA present on the ledger.
+- It can be complicated to create an interactive PoP of `CRLSignerCertificate` for some PAIs.
+- Indirect CRLs are out of scope for now
+
+## Transactions
+
+### 1. ADD_PKI_REVOCATION_DISTRIBUTION_POINT
+Publishes a PKI Revocation distribution endpoint owned by the Vendor. 
+
+If `crlSignerCertificate` is a PAA (root certificate), then it must be present on DCL.
+
+If `crlSignerCertificate` is a PAI (intermediate certificate), then it must be chained back to a valid PAA (root certificate) present on DCL.
+In this case `crlSignerCertificate` is not required to be present on DCL, and will not be added to DCL as a result of this transaction.
+If PAI needs to be added to DCL, it should be done via `ADD_X509_CERT` transaction.
+
+Publishing the revocation distribution endpoint doesn't automatically remove PAI (Intermediate certificates)
+and DACs (leaf certificates) added to DCL if they are revoked in the CRL identified by this distribution point.
+
+
+- Who can send:
+    - Vendor account
+    - `vid` field in the transaction (`VendorID`) must be equal to the Vendor account's VID
+    - `vid` field in the `CRLSignerCertificate` (for vendor-scoped PAAs and PAIs) must be equal to the Vendor account's VID
+- Parameters:
+    - vid: `uint16` -  Vendor ID (positive non-zero)
+    - pid: `optional(uint16)` -  Product ID (positive non-zero)
+    - isPAA: `bool` -  True if the revocation information distribution point relates to a PAA
+    - label: `string` -  A label to disambiguate multiple revocation information partitions of a particular issuer.
+    - crlSignerCertificate: `string` -  PEM encoded certificate (string or path to file containing data)
+    - issuerSubjectKeyID: `string` -  crlSignerCertificate's `Subject Key Id` in hex string format, e.g: `5A:88:0E:6C:36:53:D0:7F:B0:89:71:A3:F4:73:79:09:30:E6:2B:DB`
```

Comment:
Spec text is clear about explicit format:

```
For example, subject key ID A3:03:13:6D:54:A8:4B:E2:4C:48:87:B3:41:06:6D:C2:70:96:2F:99 (as it would appear in openssl x509 output, for human consumption) would be recorded as A303136D54A84BE24C4887B341066DC270962F99.
```

This does not exactly match the DCL PAA schema. Partly this is because the PAA DCL schema was "back-ported" after implementation (which made particular choices) back to spec, and the Core spec has a cross-reference to DCL implementation, which is problematic since it can move.

We have to resolve this.

The method used here (no colons) can easily renormalized for equality check with approved PAAs schema.

Reply:
Yes, thanks for noticing, I will fix it.
As of now, in order to query certificates in DCL, the SKID is expected to be with colons. So, it seems the `issuerSubjectKeyID` value will need to be split by colons, and then this new value can be used for searching PAAs on DCL as part of the checks.